### DISCUSSION TITLE
compilation target(s) code goals?

### DISCUSSION THREAD

**Discussion by Quentie** [vlang/v#11653] [Ideas]
V is looking really promising. I understand that V is intended to transpile into multiple target languages (at least c & js)? 
Quickly scanning the language I was wondering how you intend to generate low level constructs like "goto" into languages like js? 
I could imagine that most of V could be compiled into good looking js code but some features like goto are hard for me to imagine to generate human readable/understandable js code.
Will all language features be available for js target?

**Comment by Quentie** [comment]
Idea: Would it be possible to include a general purpose inline command analoge to the planned inline assembly to directly include some code in one of the compilation target languages and then even get rid of some of the commands that are hard/ugly to implement in all target languages?
```
c {
    jump:
}
...
c {
    goto jump;
}

asm x64 {
        mov eax, [a]
        add eax, 10
        mov [a], eax
}

js {
    alert('This would be neat');
}
```

**Comment by spytheman** [comment]
The js target supports (or supported before it broke):
`# console.log( message)`

**Comment by spytheman** [comment]
As for v's goto translation to JS, js supports labeled continue/break
( https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/label ).

**Comment by Quentie** [comment]
Thanks @spytheman. Did not know about # notation. This is handy! But a labeled continue/break is not really the same as a goto in c right? I mean it will still be impossible to jump to an arbitrary position in the js code but you can just break or continue outer loops right?

**Comment by spytheman** [comment]
Yes, but you can structure the generated JS code in such a way, that it is enough in most cases. V's goto is not unrestricted, just like C's goto is not unrestricted - it can not jump outside the function in which it is.

**Comment by medvednikov** [comment]
I'm still not sure about `goto` in V.

Needed it for code translated from C.

**Comment by dumblob** [comment]
>I'm still not sure about goto in V.
>Needed it for code translated from C.

What about making it reserved (including semantics - i.e. development of *V* should always consider there might be some `goto`), but not usable (i.e. disabled)?

After some time (maybe about 3 years since the first stable backwards-compatible release) we could decide to abandon it completely or enable it depending on situation (I can only think of valid `goto` uses in *V* for highly hand-optimized computation involving "**computed** gotos" - see e.g. https://internals.rust-lang.org/t/gotos-in-restricted-functions/4393/4 , but I feel it's a bit disputable in *V* as *V* offers other means to deal with extreme optimization - e.g. the compile time `$if ...` potentially useful for target/platform recognition and verbatim code writing).
